code/modules/surgery/organs/subtypes/kidan.dm

```diff
@@ -1,3 +1,105 @@
 /obj/item/organ/internal/liver/kidan
 	alcohol_intensity = 0.5
 	species = "Kidan"
+
+/obj/item/organ/internal/lantern
+	name = "bioluminescent lantern"
+	desc = "A specialized tissue that reacts with oxygen, nutriment and blood to produce light in Kidan."
+	icon_state = "kid_lantern"
+	origin_tech = "biotech=2"
+	w_class = 1
+	parent_organ = "groin"
+	slot = "lantern"
+	var/colour
+
+/obj/item/organ/internal/lantern/process()
+	if(owner.glowing)//i hate this but i couldnt figure out a better way
+		if(owner.nutrition < KIDAN_MINHUNGER)
```

Comment:
this shouldn't be a `KIDAN` define, since you can rip this out and give it to someone else.